### PR TITLE
feat: eagerly compute pruning stats during compression

### PR DESCRIPTION
**Pull request by lwwmanning**
follow up to #1236

### CODE REVIEW

**Review by robert3005 — APPROVED** [review]

**Comment by lwwmanning** [comment]
@robert3005 I'm conflicted about whether to merge this now, or perhaps bundle this in with your forthcoming stats pruning?

**Comment by robert3005** [comment]
@danking is working on stats pruning. I think you can merge it and work on stats computation improvements.

**Comment by lwwmanning** [comment]
Here's the compression benchmarks, filtered to compress time & compression ratio.

TLDR: compress time goes up by 10-50% across the board (median of ~30% increase). File size & decompress time are essentially unchanged.


| Benchmark suite | Current: [`31513b9`](https://github.com/spiraldb/vortex/commit/31513b9e1b757b3bc4ffed3459bb74028627a21d) | Previous: [`c1cee33`](https://github.com/spiraldb/vortex/commit/c1cee33288ee3dd2ad2d57209185a8eefc10c803) | Ratio |
| --- | --- | --- | --- |
| `compress time/taxi` | `1745531046.5` ns (`2489174.600000024`) | `1295362143.3` ns (`2166932.399999976`) | `1.35` |
| `compress time/AirlineSentiment` | `876210.2091746794` ns (`1997.4042628205498`) | `824226.5720406121` ns (`877.1890923065948`) | `1.06` |
| `compress time/Arade` | `3598219792.4` ns (`5449954.549999952`) | `2429903296.7` ns (`4382143.078749895`) | `1.48` |
| `compress time/Bimbo` | `16653761368` ns (`54449165.24374962`) | `11793601990.3` ns (`13185647.921250343`) | `1.41` |
| `compress time/CMSprovider` | `18126148716.4` ns (`47354529.64999962`) | `14154839403.5` ns (`16396633.357499123`) | `1.28` |
| `compress time/Euro2016` | `3149746758.7` ns (`7328379.622499943`) | `2772666256.1` ns (`2803495.6124999523`) | `1.14` |
| `compress time/Food` | `1601089528.1` ns (`2953086.517500043`) | `1214405935.4` ns (`3384823.1050001383`) | `1.32` |
| `compress time/HashTags` | `3043314601.8` ns (`4514786.694999933`) | `2718792728.4` ns (`2054576.100000143`) | `1.12` |
| `compress time/TPC-H l_comment chunked without fsst` | `4231990185` ns (`24612574.023750067`) | `3774158295.2` ns (`7205992.347500086`) | `1.12` |
| `compress time/TPC-H l_comment chunked` | `1340328374.6` ns (`5812067.313750029`) | `898236917.7` ns (`1253379.8500000238`) | `1.49` |
| `compress time/TPC-H l_comment canonical` | `1341216913.5` ns (`1761867.0193749666`) | `894183675.5` ns (`867472.6331250072`) | `1.50` |
